index.js

```diff
@@ -1,106 +1,100 @@
 'use strict'
 
 const eu = encodeURIComponent
-const figgyPudding = require('figgy-pudding')
-const getStream = require('get-stream')
 const npmFetch = require('npm-registry-fetch')
 const validate = require('aproba')
 
-const TeamConfig = figgyPudding({
-  description: {},
-  Promise: { default: () => Promise }
-})
-
 const cmd = module.exports = {}
 
-cmd.create = (entity, opts) => {
-  opts = TeamConfig(opts)
-  return pwrap(opts, () => {
+cmd.create = (entity, opts = {}) => {
+  return Promise.resolve().then(() => {
     const { scope, team } = splitEntity(entity)
     validate('SSO', [scope, team, opts])
-    return npmFetch.json(`/-/org/${eu(scope)}/team`, opts.concat({
+    const uri = `/-/org/${eu(scope)}/team`
+    return npmFetch.json(uri, {
+      ...opts,
       method: 'PUT',
       scope,
       body: { name: team, description: opts.description }
-    }))
+    })
   })
 }
 
-cmd.destroy = (entity, opts) => {
-  opts = TeamConfig(opts)
-  return pwrap(opts, () => {
+cmd.destroy = (entity, opts = {}) => {
+  return Promise.resolve().then(() => {
     const { scope, team } = splitEntity(entity)
     validate('SSO', [scope, team, opts])
-    return npmFetch.json(`/-/team/${eu(scope)}/${eu(team)}`, opts.concat({
+    const uri = `/-/team/${eu(scope)}/${eu(team)}`
+    return npmFetch.json(uri, {
+      ...opts,
       method: 'DELETE',
       scope
-    }))
+    })
   })
 }
 
-cmd.add = (user, entity, opts) => {
-  opts = TeamConfig(opts)
-  return pwrap(opts, () => {
+cmd.add = (user, entity, opts = {}) => {
+  return Promise.resolve().then(() => {
     const { scope, team } = splitEntity(entity)
     validate('SSO', [scope, team, opts])
-    return npmFetch.json(`/-/team/${eu(scope)}/${eu(team)}/user`, opts.concat({
+    const uri = `/-/team/${eu(scope)}/${eu(team)}/user`
+    return npmFetch.json(uri, {
+      ...opts,
       method: 'PUT',
       scope,
       body: { user }
-    }))
+    })
   })
 }
 
-cmd.rm = (user, entity, opts) => {
-  opts = TeamConfig(opts)
-  return pwrap(opts, () => {
+cmd.rm = (user, entity, opts = {}) => {
+  return Promise.resolve().then(() => {
```

Comment:
I think we probably don't need this Promise.resolve().then() wrapper, since the only thing that can throw is the arg validator (which we'd want to throw anyway?)

But it's harmless.  We can post a "good first contrib" issue and let someone else clean those up later :)

Reply:
In this particular case, if you had something like 
```javascript
const team = require('libnpmteam')

function main() {
  const user = '...'
  const entity = '...'
  const opts = {}
  return team.rm(user, entity, opts)
    .then((result) => {
      console.log('result:', result)
    })
    .catch((err) => {
      console.error('err:', err)
    })
}

main()
```
If `validate` blows up, there isn't a promise created yet, so you won't hit the catch that has been setup.